funcs_scripts.php

```diff
@@ -350,21 +405,24 @@ function human_cron(string $cron): string
  * between 1 and 28
  * Note that this will return the exact same value every time it is called for a given filename in a given module
  */
-function predictable_random_int($max, $offset = 0): int
+function predictable_random_int($scriptName, $max, $offset = 0): int
+{
+    $chars = str_split(module_name() . $scriptName);
+    $codes = array_map(fn($c) => ord($c), $chars);
+    $sum = array_sum($codes);
+    $remainder = $sum % ($max + 1);
+    return $remainder + $offset;
+}
+
+/**
+ * Determine if the current branch is either 1 or 1.2 style
+ */
+function current_branch_is_semver()
 {
     global $MODULE_DIR;
-    $callingFile = debug_backtrace()[0]['file'];
-    // remove absolute path e.g. /home/myuser/...
-    $moduleStandardiserDir = basename(__DIR__);
-    $dirQuoted = preg_quote($moduleStandardiserDir);
-    // double $dirQuoted is for github actions CI where there which will have a directory strcture
-    // with /module-standardiser/module-standardiser/...
-    if (!preg_match("#/$dirQuoted/$dirQuoted/(.+)$#", $callingFile, $matches)) {
-        preg_match("#/$dirQuoted/(.+)$#", $callingFile, $matches);
+    $currentBranch = cmd('git rev-parse --abbrev-ref HEAD', $MODULE_DIR);
+    if (preg_match('#^(pulls/)?([0-9]+)(\.[0-9]+)?(/|$)#', $currentBranch)) {
+        return true;
     }
```

Comment:
`pulls/*` isn't semver but this regex says it is... what's the intention here?

Reply:
The current branch could be in a `pulls/1.9/module-standardiser-12312313`, which follow our branching conventions, though I guess 'semver' is the wrong word. I've updated the function name.

Reply:
Doesn't feel super robust but we can deal with problems as they arise.